lib/task/jobs/jobWorkerTask.class.php

```diff
@@ -120,14 +165,33 @@ function ($handle, $job, $e) {
             // Another option would be to catch the ProperException from the worker
             // and restablish the connection when needed. Also, the persistent mode
             // could be disabled for this worker. See issue #4182.
-            function () use (&$counter) {
+            function ($idle, $lastJob) use (&$counter) {
+                if ($this->maxJobCountReached()) {
+                    $this->log(sprintf('Max job count reached: %u jobs completed.', $this->getJobsCompleted()), sfLogger::INFO);
+
+                    // Notify the worker that beginWork() work loop should exit.
+                    return true;
+                }
+
                 if (30 == $counter++) {
                     $counter = 0;
 
                     QubitPdo::prepareAndExecute('SELECT 1');
                 }
             }
         );
+
+        // Return code '111' when worker shuts down due to reaching max job count.
+        if ($this->getMaxJobCount() > 0 && $this->maxJobCountReached()) {
```

Comment:
`$this->getMaxJobCount() > 0` is redundant here.

Reply:
👍